### PR TITLE
Added mint test case into UniswapV2PairTest.sol, close #3

### LINKED ISSUE #3: UniswapV2Pair.spec.ts mint
rewrite tests using solidity running in Foundry.

Test descriptions:

1. mint

```
  it('mint', async () => {
    const token0Amount = expandTo18Decimals(1) // 1 000 000 000 000 000 000
    const token1Amount = expandTo18Decimals(4)
    await token0.transfer(pair.address, token0Amount)
    await token1.transfer(pair.address, token1Amount) // x , y: 1 *10 ^18 * 4 * 10 ^18

    const expectedLiquidity = expandTo18Decimals(2)
    await expect(pair.mint(wallet.address, overrides))
      .to.emit(pair, 'Transfer')
      .withArgs(AddressZero, AddressZero, MINIMUM_LIQUIDITY)
      .to.emit(pair, 'Transfer')
      .withArgs(AddressZero, wallet.address, expectedLiquidity.sub(MINIMUM_LIQUIDITY))
      .to.emit(pair, 'Sync')
      .withArgs(token0Amount, token1Amount)
      .to.emit(pair, 'Mint')
      .withArgs(wallet.address, token0Amount, token1Amount)

    expect(await pair.totalSupply()).to.eq(expectedLiquidity)
    expect(await pair.balanceOf(wallet.address)).to.eq(expectedLiquidity.sub(MINIMUM_LIQUIDITY))
    expect(await token0.balanceOf(pair.address)).to.eq(token0Amount)
    expect(await token1.balanceOf(pair.address)).to.eq(token1Amount)
    const reserves = await pair.getReserves()
    expect(reserves[0]).to.eq(token0Amount)
    expect(reserves[1]).to.eq(token1Amount)
  })
  ```

### PR DESCRIPTION
**Pull request by cirethic**
Added // SPDX-License-Identifier: UNLICENSED on all uniswap v2 source files for solc compiler 0.8.13 to be able to compile the uniswap contract files
Removed separate mint test file feature-3-mint.t.sol

### CODE REVIEW

File: test/UniswapV2PairTest.sol [hunk 1]
```diff
@@ -25,6 +32,40 @@ contract UniswapV2PairTest is Test {
...
+      emit Transfer(address(0),address(this),  expectedLiquidity - MINIMUM_LIQUIDITY);
+
+      vm.expectEmit(false, false, false, true);
+      emit Sync(uint112(token0Amount), uint112(token1Amount));
+      vm.expectEmit(true, false, false, true);
+      emit Mint(address(this), token0Amount, token1Amount);
+
+      pair.mint(address(this));
+
+      uint256 abc = expectedLiquidity;
```
Comment (duelinggalois) [inline test/UniswapV2PairTest.sol]: This line can be removed.

File: test/UniswapV2PairTest.sol [hunk 2]
```diff
@@ -13,6 +13,13 @@ contract UniswapV2PairTest is Test {
     IERC20 private token0;
     IERC20 private token1;
 
+    uint112 constant  MINIMUM_LIQUIDITY =10 ** 3;
```
Comment (duelinggalois) [inline test/UniswapV2PairTest.sol]: We should see if we can find an auto formatter to take care of missing or extra spaces. Until then, I think we should try to be consistent and aware of spacing. A few lines need an extra space or two. 

- This line between `=10`.
- line 11 `from,address`
- 58 `expectedLiquidity-MINIMUM_LIQUIDITY`

 PR looks great, just these two cosmetic suggestions.

**Review by duelinggalois — CHANGES_REQUESTED** [review]
Function wise, everything looks great. Only changes are cosmetic.

File: test/UniswapV2PairTest.sol (hunk 1 above)
Comment (cirethic) [reply test/UniswapV2PairTest.sol]: yes 👍

File: test/UniswapV2PairTest.sol (hunk 2 above)
Comment (cirethic) [reply test/UniswapV2PairTest.sol]: I was heavily relying on the auto formatter with LiteIde when coding Go. I will find a one for Atom.

**Review by duelinggalois — APPROVED** [review]

**Pull request closed without merging**